tests/regression/tests/REQUEST-942-APPLICATION-ATTACK-SQLI/942190.yaml

```diff
@@ -670,3 +670,20 @@
           version: HTTP/1.0
         output:
           no_log_contains: id "942190"
+  -
+    test_title: 942190-40
+    desc: "MSSQL Logical Functions - IIF (Transact-SQL)"
+    stages:
+    -
+      stage:
+        input:
+          dest_addr: 127.0.0.1
+          headers:
+            Host: localhost
+          method: GET
+          port: 80
+          uri: "/"
+          data: "arg=SELECT%20IIF(1%20%3E%201%2C%20'TRUE'%2C%20'FALSE'%20)"
+          version: HTTP/1.0
+        output:
+          log_contains: id "942190"
```

Comment:
I'd also like to have a test for an FP that can be observed before the change. This test is good but only covers the new state of the expression.

Reply:
This PR doesn't resolve a FP. It's more the case that the regex will never match, because a word character has to be in front of iif.
So for me the test is fine.

Reply:
I don't agree. Here's a test case that fails the *current* rule and passes the new rule (it uses a typo :) ):
```yaml
  -
    test_title: 942190-41
    desc: "False positive IIF"
    stages:
    -
      stage:
        input:
          dest_addr: 127.0.0.1
          headers:
            Host: localhost
          method: GET
          port: 80
          uri: "/"
          data: "arg=it%20is%20my%20beliif%20%28stemming%20from%20personal%20experience%29%2C%20that"
          version: HTTP/1.0
        output:
          no_log_contains: id "942190"
```

So I'd like to have that test in there to prevent a regression.

Reply:
I agree with this idea.  ;-)
It's better to add a FP test, I will update this test.

Reply:
Great, thanks! One last thing: could you change the description to match your original test case? `"MSSQL Logical Functions - IIF (Transact-SQL) - regression test"`

Reply:
Ok, no problem. ;-)